src/domain/Tables/Table/subcomponents/TableRow.tsx

```diff
@@ -267,85 +214,89 @@ const TableRow = <T extends {}>(props: ITableRowProps<T>) => {
-      if (Math.floor(useDragProps._movement[0]) !== movement) {
-        setMovement(Math.min(Math.trunc(Math.abs(useDragProps._movement[0])), swipeContentWidth))
+    if (hasSwipeActions && useDragProps._movement[0] !== 0) {
+      // new position = oldposition - movement, clamped by [0, maxwidth]
+      let newMovement = Math.floor(movement - useDragProps._movement[0])
+      newMovement = Math.max(Math.min(newMovement, swipeContentWidth), 0)
+
+      // When dragging stops, snap to one of the two ends
+      if (!useDragProps.dragging) {
+        newMovement = (newMovement < (swipeContentWidth / 2)) ? 0 : swipeContentWidth
       }
 
-      if (!useDragProps.dragging && useDragProps._movement[0] < 0) {
-        setMovement((Math.abs(useDragProps._movement[0]) < (swipeContentWidth / 2)) ? 0 : swipeContentWidth)
+      if (newMovement !== movement) {
+        setMovement(newMovement)
+
+        if (newMovement > 0) {
+          setExpandedSwipeCellRow(id)
+        } else if (newMovement === 0 && expandedSwipeCellRow !== id) {
+          setExpandedSwipeCellRow(null)
+        }
       }
     }
   })
 
-  const previousRowProps = usePrevious<IRowProps<T>>(row)
-  const previousProgressFromPreviousRowProps = previousRowProps ? (previousRowProps.progress ? previousRowProps.progress : 0) : 0
-  const previousContentOverrideFromPreviousRowProps = previousRowProps && previousRowProps.contentOverride
-  const hasContentOverrideChange = !!previousRowProps && !isEqual(previousContentOverrideFromPreviousRowProps, row.contentOverride)
-  const isSelected = hasLeftAction && isSelectable ? selectedRows[row.id] : false
-  const handleSwipeAction = () => hasSwipeActions ? (movement !== 0 ? setMovement(0) : setMovement(swipeContentWidth)) : undefined
-  const hasHoverButton = (selectedAll === TableCheckboxInputValue.False && hasHoverActions) ? hasHovered : false
-
   useEffect(() => {
-    // get previous progress when progress is given and different from the current progress
-    if (progress && previousProgressFromPreviousRowProps !== (progress ? progress : 0)) {
-      setPreviousProgress(previousProgressFromPreviousRowProps)
-    }
-  }, [progress])
-  useEffect(() => {
-    // when user expands this row swipe cell, set expanded row id as this row id to force other row swipe cell collapse
-    if (movement > 0) {
-      setExpandedSwipeCellRow(id)
-    }
-
-    // when user collapse this row swipe cell (not when it is force collapsed by other row), set expanded row id null
-    if (movement === 0 && !(expandedSwipeCellRow !== null && expandedSwipeCellRow !== id)) {
-      setExpandedSwipeCellRow(null)
-    }
-  }, [movement])
-  useEffect(() => {
-    // when user expands other row, collapse this row
-    if (expandedSwipeCellRow !== null && expandedSwipeCellRow !== id) {
+    if (expandedSwipeCellRow !== id) {
       setMovement(0)
     }
   }, [expandedSwipeCellRow])
```

Comment:
effects were moved to `handleSwipeAction` and `onDrag`, which causes only a single rerender rather than the loops.